packages/sdk/src/index.ts

```diff
@@ -99,21 +83,75 @@ export class WHIPClient {
     this.log("IceCandidateError", e);
   }
 
-  async ingest(mediaStream: MediaStream): Promise<void> {
-    if (this.opts.iceConfigFromEndpoint) {
-      const iceServers: WHIPClientIceServer[] = await this.doFetchICEFromEndpoint();
-      this.peer.setConfiguration({ iceServers: iceServers });
+  private onIceGatheringTimeout() {
+    this.log("IceGatheringTimeout");
+    clearTimeout(this.iceGatheringTimeout);
+
+    this.peer.removeEventListener("icecandidate", this.onIceCandidate.bind(this));
+    this.onIceGatheringComplete();
+  }
+
+  private async onIceGatheringComplete() {
+    if (this.iceGatheringComplete) {
+      return;
     }
+    this.log("IceGatheringComplete");
 
-    mediaStream
-      .getTracks()
-      .forEach((track) => this.peer.addTrack(track, mediaStream));
+    this.iceGatheringComplete = true;
+
+    const response = await fetch(this.whipEndpoint.toString(), {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/sdp",
+        "Authorization": this.opts.authkey
+      },
+      body: this.peer.localDescription.sdp,
+    });
+
+    if (response.ok) {
+      this.resource = response.headers.get("Location");
+      this.log("WHIP Resource", this.resource);
+      if (this.resourceResolve) {
+        this.resourceResolve(this.resource);
+        this.resourceResolve = null;
+      }
+
+      const answer = await response.text();
+      this.peer.setRemoteDescription({
+        type: "answer",
+        sdp: answer,
+      });
+    } else {
+      this.error("IceCandidate", "Failed to setup stream connection with endpoint", response.status, await response.text());
+    }
+  }
 
+  private async startSdpExchange(): Promise<void> {
     const sdpOffer = await this.peer.createOffer({
       offerToReceiveAudio: false,
       offerToReceiveVideo: false,
     });
     this.peer.setLocalDescription(sdpOffer);
+
+    this.peer.addEventListener("icecandidate", this.onIceCandidate.bind(this));
+    this.iceGatheringComplete = false;
+    this.iceGatheringTimeout = setTimeout(this.onIceGatheringTimeout.bind(this), this.opts.iceGatheringTimeout || DEFAULT_ICE_GATHERING_TIMEOUT);
+  }
+
+  async init(): Promise<void> {
```

Comment:
Unless we have other plans for `init()` I think it would make sense to remove the `iceConfigFromEndpoint` option and instead go with a method `connectToWhipIceServers` or `setIceServersFromEndpoint`, the option is a bit redundant at this point.

Reply:
Fair point! Will make a setIceServersFromEndpoint